tests/zkevm/test_worst_blocks.py

```diff
@@ -0,0 +1,160 @@
+"""
+abstract: Tests zkEVMs worst-case block scenarios.
+    Tests zkEVMs worst-case block scenarios.
+
+Tests running worst-case block scenarios for zkEVMs.
+"""
+
+import pytest
+
+from ethereum_test_forks import Fork
+from ethereum_test_tools import (
+    Account,
+    Address,
+    Alloc,
+    Block,
+    BlockchainTestFiller,
+    Environment,
+    Transaction,
+)
+
+
+@pytest.fixture
+def iteration_count(eth_transfer_cost: int):
+    """Calculate the number of iterations based on the gas limit and intrinsic cost."""
+    return Environment().gas_limit // eth_transfer_cost
+
+
+@pytest.fixture
+def transfer_amount():
+    """Ether to transfer in each transaction."""
+    return 1
+
+
+@pytest.fixture
+def eth_transfer_cost(fork: Fork):
+    """Transaction gas limit."""
+    intrinsic_cost = fork.transaction_intrinsic_cost_calculator()
+    return intrinsic_cost()
+
+
+def get_distinct_sender_list(pre: Alloc):
+    """Get a list of distinct sender accounts."""
+    while True:
+        yield pre.fund_eoa()
+
+
+def get_distinct_receiver_list(pre: Alloc):
+    """Get a list of distinct receiver accounts."""
+    while True:
+        yield pre.fund_eoa(0)
+
+
+def get_single_sender_list(pre: Alloc):
+    """Get a list of single sender accounts."""
+    sender = pre.fund_eoa()
+    while True:
+        yield sender
+
+
+def get_single_receiver_list(pre: Alloc):
+    """Get a list of single receiver accounts."""
+    receiver = pre.fund_eoa(0)
+    while True:
+        yield receiver
+
+
+@pytest.fixture
+def ether_transfer_case(
+    request,
+    pre: Alloc,
+):
+    """Generate the test parameters based on the case ID."""
+    case_id = request.param
+
+    if case_id == "a_to_a":
+        """Sending to self."""
+        senders = get_single_sender_list(pre)
+        receivers = get_single_receiver_list(pre)
```

Comment:
```suggestion
        senders = get_single_sender_list(pre)
        receivers = get_single_receiver_list(pre)
```
```suggestion
        senders = get_single_sender_list(pre)
        receivers = senders
```
I think we have to return the exact same generator twice here, otherwise is going to return two different addresses.